### PR TITLE
Adjust operator functional tests to custom images specification

### PR DESCRIPTION
**Pull request by iholder101**
**What this PR does / why we need it**:
Since Kubevirt now supports any custom image format (see https://github.com/kubevirt/kubevirt/pull/8673) the functional tests need to allow such formats.

In this commit, parseImage() function is changed to adjust to these new capabilities.

I've made the following examples to demonstrate how the new logic works:
see this example for `@sha256` version format: https://go.dev/play/p/oRmW0MqIfgp
see this example for `:my-tag` version format: https://go.dev/play/p/d_tjdC7-0KX

**Special notes for your reviewer**:
See https://github.com/kubevirt/kubevirt/pull/8673, https://github.com/kubevirt/kubevirt/pull/8792, https://github.com/kubevirt/kubevirt/pull/8811, https://github.com/kubevirt/kubevirt/pull/8887 for previous PRs in this area.

**Release note**:

```release-note
Adjust operator functional tests to custom images specification
```

### CODE REVIEW

**Review by dollierp — APPROVED** [review]

**Comment by iholder101** [comment]
/cc @jean-edouard 

Can you also please take a look?

File: tests/operator_test.go
Comment (jean-edouard) [inline tests/operator_test.go:535]:
```suggestion
				imageRegEx = regexp.MustCompile(`^(.+)/(.+)(@sha\d+:)([\da-fA-F]+)$`)
```
- `*` matches 0 character, which we probably don't want here
- sha types are always numerical values AFAIK
- shas are always hex

File: tests/operator_test.go
Comment (jean-edouard) [inline tests/operator_test.go:539]:
```suggestion
				imageRegEx = regexp.MustCompile(`^(.+)/(.+)(:.+)$`)
```

File: tests/operator_test.go
Comment (jean-edouard) [inline tests/operator_test.go:546]: Not sure I see the point of all the parenthesis in the regexps and the check that they all matched...
If the string matched the regexp, all the groups in parentheses got matched too, even if only by the empty string...
But I guess it doesn't hurt anything, just thinking out loud.

**Review by jean-edouard — COMMENTED** [review]
/approve
/hold
The regexps can probably be improved, see in-line. Feel free to un-hold if/when fixed.
Thanks!

File: tests/operator_test.go
Comment (iholder101) [reply tests/operator_test.go]: Thanks a lot! Makes total sense!

File: tests/operator_test.go
Comment (iholder101) [reply tests/operator_test.go]: :+1:

File: tests/operator_test.go
Comment (iholder101) [reply tests/operator_test.go]: Yup. This logic existed previously, but in a second though it looks like it's redundant.
Thanks!

**Comment by iholder101** [comment]
Thanks a lot @jean-edouard! Great insights!

You're welcome to unhold and LGTM if there are no further comments :)

**Comment by jean-edouard** [comment]
/hold cancel
My comments are addressed, thank you!
Since I /approved, I'd like someone else to LGTM, another set of eyes on this can't hurt!

**Comment by iholder101** [comment]
> /hold cancel My comments are addressed, thank you! Since I /approved, I'd like someone else to LGTM, another set of eyes on this can't hurt!

You're right, it's just that it was LGTMed before by someone else :)
@dollierp would you mind taking another look?

**Comment by dhiller** [comment]
/override pull-kubevirt-fossa

This PR is not the reason FOSSA check started failing, overriding to unblock it

**Pull request merged**
